Lesson_22/js/selector.js

```diff
@@ -0,0 +1,64 @@
+export class Selector {
+   #target = null;
+   #lastSelectedIndex = null;
+
+   constructor(list) {
+      this.list = list;
+      this.items = list.children;
+
+      list.addEventListener('click', this.onClick.bind(this));
+      list.addEventListener('mousedown', e => e.preventDefault());
+   }
+
+   #clearSelected() {
+      for (let li of this.items) {
+         li.classList.remove('selected');
+      }
+   }
+
+   #addSelected(target) {
+      target.classList.add('selected');
+   }
+
+   #getIndex(item) {
+      return [...this.items].indexOf(item);
+   }
+
+   onClick(e) {
+      if (e.target.tagName !== "LI") return;
+
+      this.#target = e.target;
+
+      if (e.ctrlKey || e.metaKey) {
+         this.toggleSelected();
+      } else if (e.shiftKey) {
+         this.selectRange();
+      } else {
+         this.singleSelect();
+      }
+
+      this.#lastSelectedIndex = this.#getIndex(this.#target);
+   }
+
+   toggleSelected() {
+      this.#target.classList.toggle('selected');
+   }
+
+   singleSelect() {
+      this.#clearSelected();
+      this.#addSelected(this.#target);
+   }
+
+   selectRange() {
+      if (this.list.querySelectorAll('.selected').length === 0) {
```

Comment:
this expression can be shortened as 
```javascript
if (!this.list.querySelectorAll('.selected').length)
```
because 0 its a falsy value
https://developer.mozilla.org/en-US/docs/Glossary/Falsy